packages/backend-openapi-utils/src/testUtils.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Express } from 'express';
+import { Server } from 'http';
+
+/**
+ * Running against supertest, we need some way to hit the optic proxy. This ensures that
+ *  that happens at runtime when in the context of a `yarn optic capture` command.
+ * @param app - Express router that would be passed to supertest's `request`.
+ * @returns A wrapper around the express router (or the router untouched) that still works with supertest.
+ * @public
+ */
+export const createSuperTestAgent = (app: Express): Server | Express => {
```

Comment:
This isn't really specific to `supertest` right? Thinking in particular about the naming of this function. Perhaps something like `wrapInOpenApiTestServer`?

Reply:
Hmm, this is somewhat tied to the supertest underlying implementation, ie getting the `address()` of the http server. While that may be the same for other testing libraries, I'd like to call it out that it's just supertest compatible. How about 
```
/**
  * Only compatible with supertest.
**/
const wrapInOpenApiTestServer = () => {}
```

Reply:
Yep, I'm thinking a bit further into the future as well where if we were to add support for other harnesses I'd be nice to just add that directly to this same function too. It's just a node http server in the end so it should be possible to properly shim support for any harness that works towards that I would think.

Reply:
Makes sense, I don't think we can check which harness we're running in, but assuming harnesses don't vary _too_ much.